cabal-install/Distribution/Client/Init.hs

```diff
@@ -207,8 +207,22 @@ getLicense flags = do
          ?>> fmap (fmap (either UnknownLicense id))
                   (maybePrompt flags
                     (promptList "Please choose a license" listedLicenses (Just BSD3) display True))
-  return $ flags { license = maybeToFlag lic }
+
+  invalidOtherLicense <- if isLicenseInvalid lic
```

Comment:
Stylistic note: You don't need this extra bind: you can just move the `promptInvalidOtherLicenseMsg` printing to the `invalidOtherLicense` test. (P.S. The reason `getPackageName` has this bind is because it actually needs to ask the user to find out if the boolean will be true or false. In your case, it's always True, so it's not necessary.